### DISCUSSION TITLE
Issues on moveit/issues unresolved

### DISCUSSION THREAD

**Discussion by syedshazli** [moveit/moveit2#3186] [General]
Hi everyone, I was looking to contribute to moveit2, but this is my first time contributing to Open-Source projects. I have extensive C++ experience but being this project involves hardware/robotics, what would be the most effective way to test issues in the moveit2/issues tab?

**Comment by sea-bass** [comment]
I would start by setting up a source build of MoveIt with these instructions:
https://moveit.picknik.ai/main/doc/tutorials/getting_started/getting_started.html

Once there, you should be able to dive into the code, rebuild, and test on "mock hardware" simulated robots as the tutorials show.

**Comment by syedshazli** [comment]
Thanks so much! I'll start right now
